packages/fastify/src/ByuJwtProvider.ts

```diff
@@ -0,0 +1,56 @@
+import type { FastifyPluginAsync, FastifyReply, FastifyRequest } from 'fastify'
+import fp from 'fastify-plugin'
+import type { ByuJwtOptions, JwtPayload } from '@byu-oit/jwt'
+import { ByuJwtAuthenticator } from './ByuJwtAuthenticator.js'
+
+/** Enhance the fastify request with the verified caller information */
+declare module 'fastify' {
+  interface FastifyRequest {
+    caller: JwtPayload | null
+  }
+}
+
+export interface ByuJwtProviderOptions extends ByuJwtOptions {
+  prefix?: string
+}
+
+const ByuJwtProviderPlugin: FastifyPluginAsync<ByuJwtProviderOptions> = async (fastify, options) => {
+  const authenticator = new ByuJwtAuthenticator(options)
+
+  async function ByuJwtAuthenticationHandler (request: FastifyRequest, reply: FastifyReply): Promise<void> {
+    try {
+      request.caller = await authenticator.authenticate(request.headers)
+    } catch (err) {
+      void reply.code(401)
+      throw err
```

Comment:
In that migration documentation (issue?), we'll need to document that this no longer sends a response, like the old middleware we had for Express.

Reply:
Correct, it requires setting an error handler in fastify that catches hook errors. This is how Fastify recommends doing it: https://www.fastify.io/docs/latest/Reference/Hooks/#manage-errors-from-a-hook